prov/verbs/src/verbs_mr.c

```diff
@@ -301,25 +275,43 @@ void fi_ibv_mr_cache_delete_region(struct ofi_mr_cache *cache,
 		(void)ibv_dereg_mr(md->mr);
 }
 
-static int fi_ibv_mr_cache_regattr(struct fid *fid, const struct fi_mr_attr *attr,
-				   uint64_t flags, struct fid_mr **mr)
+static int
+fi_ibv_mr_cache_reg(struct fid *fid, const void *buf, size_t len,
+		    uint64_t access, uint64_t offset, uint64_t requested_key,
+		    uint64_t flags, struct fid_mr **mr, void *context)
 {
 	struct fi_ibv_domain *domain;
 	struct fi_ibv_mem_desc *md;
-	int ret;
+	struct fi_mr_attr attr;
+	struct iovec iov;
 
-	ret = fi_ibv_mr_regattr_check_args(fid, attr, flags);
-	if (OFI_UNLIKELY(ret))
-		return ret;
+	if (OFI_UNLIKELY(flags))
+		return -FI_EBADFLAGS;
 
 	domain = container_of(fid, struct fi_ibv_domain,
 			      util_domain.domain_fid.fid);
 
-	md = fi_ibv_mr_common_cache_reg(domain, (struct fi_mr_attr *)attr);
+	attr.access = access;
+	attr.context = context;
+	attr.iov_count = len ? 1 : 0;
```

Comment:
I'm seeing the following assertion fail which is probably due to this change:
https://github.com/ofiwg/libfabric/blob/ea11792f6119d714806df4c7af0b48b397172b6b/prov/util/src/util_mr_cache.c#L282

Reply:
That assertion has always been there.  I thought verbs was handling empty regions because MPI was requesting them for some goofy reason.  Anyway, I'm trying to understand how this was working.  Is it expecting an iov with a len of 0 for an empty region?  That could make sense based on how the code was originally structured.

Reply:
Verbs handles empty regions but it handles it in a function that receives the buf and len directly and not in an iovec. The change in this PR updates the count based on length which ends up failing the assertion in the cache.